gio/src/dbus_connection.rs

```diff
@@ -28,80 +28,83 @@ impl DBusConnection {
         &self,
         object_path: &str,
         interface_info: &DBusInterfaceInfo,
-        method_call: MethodCall,
-        get_property: GetProperty,
-        set_property: SetProperty,
+        method_call: Option<MethodCall>,
+        get_property: Option<GetProperty>,
+        set_property: Option<SetProperty>,
```

Comment:
Did you try to pass a `None` here? You'll notice that's impossible because you can't name any type for the closure :)

It would either have to be a `Option<Box<Fn ...>>` or you need a separate variant of the function that always uses no closures (but considering that there are 3 here, adding 9 functions seems not very useful...)

Reply:
Could make it use a builder pattern :) similar to how we do for properties bindings

Reply:
Added the builder